File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-33.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var set = new HashSet<char>();
        var l = 0;
        var res = 0;
        for (int r = 0; r < s.Length; r++) {
            while (set.Contains(s[r])) {
                set.Remove(s[l]);
                l++;
            }
            set.Add(s[r]);
            res = Math.Max(res, r - l + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-33.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        l = 0
        for r in range(len(prices)):
            if prices[r] > prices[l]:
                res = max(prices[r] - prices[l], res)
            else:
                l = r
        return res